### PR TITLE
[SYCL][DebugInfo] Switch to nonsemantic-shader-200 for non-FPGA HW on…

### PR DESCRIPTION
**Pull request by MrSidims**
… linux

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -10218,10 +10218,23 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
...
-    // operations which don't have mapping to OpenCL.DebugInfo.100 spec.
-    TranslatorArgs.push_back("-spirv-allow-extra-diexpressions");
+    // Enable NonSemanticShaderDebugInfo.200 for CPU AOT and for non-Windows
+    // JIT. Don't enable on FPGA H/W.
+    const llvm::Triple *AuxTriple = TC.getAuxTriple();
+    const bool IsWindowsMSVC =
+        Triple.isWindowsMSVCEnvironment() ||
+        (AuxTriple && AuxTriple->isWindowsMSVCEnvironment());
+    const bool EnableNonSemanticDebug =
+        IsCPU || (!IsWindowsMSVC && !C.getDriver().IsFPGAHWMode());
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp:10228]: Comment states 'non-Windows JIT'.  This check allows for all non-windows, including AOT.  Is that OK?

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:10228]: Apologies to the long response. Yes, non-Windows AOT is also OK until it's not FPGA.

**Review by mdtoguchi — APPROVED** [review]

**Comment by MrSidims** [comment]
holding the merge a bit

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers please help with merge

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers please help with the merge

**Comment by againull** [comment]
@MrSidims It looks like this commit caused unexpected passes in post-commit:
********************
Unexpectedly Passed Tests (2):
  SYCL :: InvokeSimd/Regression/ImplicitSubgroup/debug_symbols.cpp
  SYCL :: InvokeSimd/Regression/debug_symbols.cpp

https://github.com/intel/llvm/actions/runs/8723664071
Could you please fix that.

**Comment by MrSidims** [comment]
@againull @sarnex sure, I'll take a look, thanks for the notice!

**Pull request merged**